### PR TITLE
Fix tab-completion in remote shell by setting default-directory

### PR DESCRIPTION
**Pull request by bowenli37**
ref: http://stackoverflow.com/questions/4683256/tab-completion-in-emacs-shell-mode-ssh-sessions

### CODE REVIEW

**Comment by ieure** [comment]
This will break if you pass other arguments to SSH (ex `-i some_identity_file`) when you invoke `M-x ssh`.

I haven't experienced this issue. I have this stuff in my init files:

``` emacs-lisp
(eval-after-load 'ssh
  '(progn
     (add-hook 'ssh-mode-hook
               (lambda ()
                 (shell-dirtrack-mode t)
                 (setq dirtrackp nil)))))
```

Does that solve the issue for you?

**Comment by bowenli37** [comment]
Thank you very much. It works perfect.

Ian Eure notifications@github.com writes:

> This will break if you pass other arguments to SSH (ex -i some_identity_file)
> when you invoke M-x ssh.
> 
> I haven't experienced this issue. I have this stuff in my init files:
> 
> (eval-after-load 'ssh
>   '(progn
>      (add-hook 'ssh-mode-hook
>                (lambda ()
>                  (shell-dirtrack-mode t)
>                  (setq dirtrackp nil)))))
> 
> Does that solve the issue for you?
> 
> —
> Reply to this email directly or view it on GitHub.*

## 

Sincerely,
Li Bowen.

**Comment by bowenli37** [comment]
Hi,

I did more tests and found that putting the snippet in .init.el does not
solve the problem. default-directory has to be set to
"/ssh:user@host:/". But dirtracking will fail once absolute path is
used, like cd / or cd ~.

Thanks.

Ian Eure notifications@github.com writes:

> This will break if you pass other arguments to SSH (ex -i some_identity_file) when you invoke M-x ssh.
> 
> I haven't experienced this issue. I have this stuff in my init files:
> 
> (eval-after-load 'ssh
>   '(progn
>      (add-hook 'ssh-mode-hook
>                (lambda ()
>                  (shell-dirtrack-mode t)
>                  (setq dirtrackp nil)))))
> 
> Does that solve the issue for you?
> 
> —
> Reply to this email directly or view it on GitHub.*

## 

Sincerely,
Li Bowen.

File: ssh.el [hunk 1]
```diff
@@ -260,6 +260,10 @@ how ssh X display tunelling interacts with frames on remote displays."
                 (t
                  (cd-absolute (concat comint-file-name-prefix "~/"))))
         (error nil)))))
+
+  ;; enable tab-complete in remote shell
+  (setq default-directory (concat "/ssh:" input-args ":~/"))
```
Comment (zenspider) [inline ssh.el:265]: This should probably use `tramp-default-method` or similar instead of hard-coding "ssh".

**Pull request closed without merging**